lib/cmd/compile/templates.js

```diff
@@ -0,0 +1,215 @@
+ * so they can be rendered client-side
+ * note: this only replaces straight file reads, not reads from dynamic filepaths
+ * note: we are explicitly ignoring clay-kiln, as it has other logic for inlining icons
+ * @param  {string} source
+ * @param  {string} filepath
+ * @return {string}
+ */
+function inlineRead(source, filepath) {
+  const staticIncludes = source.match(/\{\{\{\s?read\s?'(.*?)'\s?\}\}\}/ig),
+    name = _.last(path.dirname(filepath).split(path.sep));
+
+  let inlined = source;
+
+  _.each(staticIncludes, function (match) {
+    const filepath = match.match(/'(.*?)'/)[1];
+
+    let fileContents;
+
+    try {
+      fileContents = escape(fs.readFileSync(filepath, 'utf8')); // read file, then escape any single-quotes
+    } catch (e) {
+      console.log(chalk.red(`Error replacing {{{ read \'${filepath}\' }}} in "${name}": `) + e.message);
+      process.exit(1);
+    }
+
+    inlined = inlined.replace(match, fileContents);
+  });
+  return inlined;
+}
+
+/**
+ * wrap templates so they don't render without data, see https://github.com/clay/handlebars/blob/master/index.js#L45
+ * @param  {Vinyl} file
+ * @return {Vinyl}
+ */
+function wrapTemplate(file) {
+  let source = _.includes(file.path, 'clay-kiln') ? file.contents.toString('utf8') : inlineRead(file.contents.toString('utf8'), file.path);
+
+  file.contents = new Buffer(clayHbs.wrapPartial(_.last(path.dirname(file.path).split(path.sep)), source));
+  return file;
+}
+
+/**
+ * precompile handlebars templates into js functions
+ * @param  {Vinyl} file
+ * @return {Vinyl}
+ */
+function precompile(file) {
+  const name = path.parse(file.path).name.replace('.template', '');
+
+  try {
+    file.contents = new Buffer(hbs.precompile(file.contents.toString('utf8')));
+    return file;
+  } catch (e) {
+    console.log(chalk.red(`Error precompiling template "${name}": `) + e.message);
+    process.exit(1);
+  }
+}
+
+/**
+ * register templates by adding them to the 'window' object
+ * @param  {Vinyl} file
+ * @return {Vinyl}
+ */
+function registerTemplate(file) {
+  const name = path.parse(file.path).name.replace('.template', ''),
+    contents = file.contents.toString('utf8');
+
+  file.contents = new Buffer(`window.kiln.componentTemplates['${name}']=${contents}\n`);
+  return file;
+}
+
+/**
+ * minify template js, if 'minify' argument is passed in
+ * @param  {Vinyl} file
+ * @param  {boolean} shouldMinify
+ * @return {Vinyl}
+ */
+function minifyTemplate(file, shouldMinify) {
+  if (!shouldMinify) {
+    // don't do anything, pass it through
+    return file;
+  }
+
+  try {
+    const minified = uglify.minify(file.contents.toString('utf8'), { output: { inline_script: true } });
+
+    file.contents = new Buffer(minified.code);
+    return file;
+  } catch (e) {
+    const name = path.parse(file.path).name.replace('.template', '');
+
+    console.log(chalk.red(`Error minifying template "${name}": `) + e.message);
```

Comment:
`.error`

Reply:
I'm using `.log()` here (and in other places where compilation errors might happen) so it'll write to `stdout`